packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -40,6 +40,7 @@ abstract class UnpackMacOS extends Target {
   @override
   List<Source> get outputs => const <Source>[
     Source.pattern('{OUTPUT_DIR}/FlutterMacOS.framework/Versions/A/FlutterMacOS'),
+    Source.pattern('{OUTPUT_DIR}/FlutterMacOS.framework.dSYM/Contents/Resources/DWARF/FlutterMacOS'),
```

Comment:
This won't exist for non-release versions, so I believe this would make the caching fail and this target would be run every time, because the output isn't present?

I  think you'd want it in `ReleaseUnpackMacOS`
https://github.com/flutter/flutter/blob/aa934ac119d007ad2e9dac7dbd75e41c1ee17ac8/packages/flutter_tools/lib/src/build_system/targets/macos.dart#L159-L162

And same for iOS:
https://github.com/flutter/flutter/blob/aa934ac119d007ad2e9dac7dbd75e41c1ee17ac8/packages/flutter_tools/lib/src/build_system/targets/ios.dart#L409-L410

Reply:
Oh crap -- yes I was going to ask about this. Here's the comment from `git reflog` before I squashed the commits:

```
commit cdad6140d912abff055b622563ef70fd6b3a7693
Author: Chris Bracken <chris@bracken.jp>
Date:   Thu Aug 22 11:48:47 2024 -0700

    Add macOS dSYM copy operation

    Can I really just add the dSYM to the outputs list? Won't debug/profile
    builds complain that the output wasn't produced? Maybe our build system
    doesn't check?
```

Reply:
Done.